tools/codegen/static_runtime/gen_structured.py

```diff
@@ -0,0 +1,310 @@
+import tools.codegen.api.cpp as cpp
+from tools.codegen.context import native_function_manager
+from tools.codegen.model import (Argument, BaseTy, FunctionSchema, OptionalType,
+                                 SelfArgument,
+                                 BaseType, NativeFunctionsGroup, TensorOptionsArguments, Type)
+from tools.codegen.static_runtime import config
+
+import math
+from typing import List, Optional, Sequence, Tuple, Union
+
+
+def has_alias(arguments: Sequence[Union[Argument, SelfArgument, TensorOptionsArguments]]) -> bool:
+    for arg in arguments:
+        annotation = getattr(arg, "annotation", None)
+        if not annotation:
+            continue
+        alias_set = getattr(annotation, "alias_set", ())
+        if alias_set:
+            return True
+    return False
+
+def is_supported(g: NativeFunctionsGroup) -> bool:
+    if not g.structured:
+        return False
+    if config.is_hand_written(g):
+        return False
+    if has_alias(g.out.func.arguments.non_out):
+        # This op may create an alias of inputs.
+        return False
+    if len(g.out.func.arguments.out) > 1:
+        # More than 1 output values.
+        return False
+    if "at::Tensor &" != cpp.returns_type(g.out.func.returns).cpp_type():
+        # Returns a non-Tensor value.
+        return False
+    for arg in g.out.func.schema_order_arguments():
+        maybe_method = ivalue_type_conversion_method(arg.type)
+        if not maybe_method:
+            # Type converting is unsupported yet.
+            return False
+    return True
+
+def ivalue_type_conversion_method(arg_type: Union[BaseType, OptionalType, Type]) -> Optional[Tuple[bool, str]]:
+    """
+    Return the method call expression of `c10::ivalue' to convert its contained value to
+    the expected value of `arg_type` type. For example, for `arg_type` == BaseTy.Tensor,
+    this function returns ".toTensor()", so that it can be appended to the ivalue's
+    variable name to get the value of the expected type.
+    """
+    type_conversion_methods = {
+        BaseTy.Tensor: ((True, "toTensor()"), (False, "toOptional<at::Tensor>()")),
+        BaseTy.int: ((False, "toInt()"), (False, "toOptional<int64_t>()")),
+        BaseTy.bool: ((False, "toBool()"), (False, "toOptional<bool>()")),
+        BaseTy.Scalar: ((False, "toScalar()"), (False, "toOptional<at::Scalar>()")),
+        BaseTy.ScalarType: ((False, "toScalarType()"), (False, "toOptional<at::ScalarType>()")),
+        BaseTy.str: ((False, "toStringView()"), (False, "toOptional<c10::string_view>()"))}
```

Comment:
@larryliu0820 we were just talking about how IValue to C++ value conversions could be usefully used in other contexts, and lo and behold, here is some new dispatcher code which also is implementing IValue conversion code.

Reply:
(unfortunately not in a modular way either; boo)